examples/llm/src/models/mosaic_gpt.py

```diff
@@ -315,11 +368,17 @@ def get_targets(self, batch):
         return targets
 
     def forward(self, batch):
+        if self.model.prefix_lm:
```

Comment:
The answer may be no (and I haven't been able to think of one), but is there anywhere we can put this logic other than here? Trying to avoid custom logic in `ComposerMosaicGPT` if possible.

Reply:
This same logic is actually used by our HF prefix LM. It's mostly a workaround to support ICL eval for prefix LMs.
We can remove these logics once the rest of our upstream stack embraces prefix masks... Someday...

Reply:
That is the answer I expected, sounds good :)